cl/lib/email_backends.py

```diff
@@ -0,0 +1,235 @@
+                        from_email=from_email,
+                        to=to,
+                        bcc=bcc,
+                        cc=cc,
+                        reply_to=reply_to,
+                        subject=subject,
+                        message=plaintext_body,
+                        html_message=html_body,
+                        headers=headers,
+                    )
+
+                    # Compose normal messages with attachments
+                    # according to available content types.
+                    # Discard text/plain_small and text/html_small
+                    # content types.
+
+                    email = compose_message(
+                        html_body=html_body,
+                        plain_body=plaintext_body,
+                        from_email=from_email,
+                        to=to,
+                        bcc=bcc,
+                        cc=cc,
+                        reply_to=reply_to,
+                        subject=subject,
+                        headers=headers,
+                    )
+
+                    # Add attachments
+                    for attachment in email_message.attachments:
+                        # An attachment is a tuple:
+                        # (filename, content, mimetype)
+                        email.attach(
+                            attachment[0], attachment[1], attachment[2]
+                        )
+
+            else:
+                # Store message in DB and obtain the unique message_id to
+                # add in headers to identify the message
+                stored_id = store_message(
+                    from_email=from_email,
+                    to=to,
+                    bcc=bcc,
+                    cc=cc,
+                    reply_to=reply_to,
+                    subject=subject,
+                    message=plaintext_body,
+                    html_message=html_body,
+                    headers=headers,
+                )
+                # If not small version, send original message.
+                email = email_message
+
+            # Use backend connection
+            email.connection = connection
+            # Add header to with the unique message_id
+            email.extra_headers["X-CL-ID"] = stored_id
+
+            # Verify if recipient email address is under a backoff event
+            backoff_event = BackoffEvent.objects.filter(
+                email_address=convert_list_to_str(to),
+            ).first()
+
+            # If backoff event exists, check if it's under waiting period
+            if backoff_event.under_waiting_period if backoff_event else False:
+                # TODO QUEUE email
+                pass
+            else:
+                # If not under backoff waiting period, continue sending.
+                try:
+                    email.send()
+                except Exception as e:
+                    # Log an error
+                    logger.error(f"Error sending email: {e}")
```

Comment:
I think I'd rather let these crash. We'll get Sentry reports automatically that we can decide what to do with.

Reply:
Great, changed!